src/components/CellsContainer/CellsContainer.svelte

```diff
@@ -0,0 +1,201 @@
+<script>
+  import { onMount, onDestroy } from 'svelte';
+  import { select } from 'd3';
+  import { createSvg, getId, createRectsData, setAttributes } from './helpers';
+
+  export let count = 2;
+  export let size = 100;
+  export let gap = 1;
+  export let getAdditionalRectProps;
+  export let tooltipBackground = '#4a5568';
+  export let tooltipColor = 'MintCream';
+  export let color = 'SteelBlue';
+  export let hoverColor = 'IndianRed';
+  export let margin = { top: 20, left: 20, right: 20, bottom: 20 };
+
+  let containerRef = null;
+  let svg = null;
+  let g = null;
+  let data = [...Array(count)];
+  let containerWidth = 0;
+  let height = 0;
+  let innerWidth = 0;
+
+  $: {
+    if (containerWidth) {
+      innerWidth = containerWidth - margin.left - margin.right;
+
+      const rectSize = size + gap * 2;
+      const colsAmount = Math.floor(innerWidth / rectSize);
+      const rowsAmount = Math.ceil(count / colsAmount);
+
+      height = rectSize * rowsAmount + margin.top + margin.bottom;
+
+      data = createRectsData({
+        containerWidth: innerWidth,
+        count,
+        size,
+        gap,
+        getAdditionalRectProps,
+        color,
+        hoverColor
+      });
+
+      updateSvgElementsSizes();
+      updateRects();
+    }
+  }
+
+  const keyGetter = (key) => (r) => r[key];
+
+  onMount(() => {
+    svg = createSvg(containerWidth, height);
+    g = svg.append('g');
+
+    containerRef.appendChild(svg.node());
+
+    g.selectAll('rect')
+      .data(data)
+      .enter()
+      .append('rect')
+      .attr('rx', 3)
+      .attr('ry', 3)
+      .style('cursor', 'pointer')
+      .on('mouseover', handleRectHover)
+      .on('mouseout', handleRectMouseout);
+  });
+
+  function updateSvgElementsSizes() {
+    svg && svg.attr('width', containerWidth).attr('height', height);
+    g && g.style('transform', `translate(${margin.left}px, ${margin.top}px)`);
+  }
+
+  function updateRects() {
+    if (g) {
+      const rects = g.selectAll('rect').data(data);
+      rects.exit().remove();
+      rects
+        .enter()
+        .append('rect')
+        .on('mouseover', handleRectHover)
+        .on('mouseout', handleRectMouseout)
+        .merge(rects)
+        .attr('fill', keyGetter('color'))
+        .attr('rx', 3)
+        .attr('ry', 3)
+        .attr('x', keyGetter('x'))
+        .attr('y', keyGetter('y'))
+        .attr('width', keyGetter('size'))
+        .attr('height', keyGetter('size'))
+        .style('cursor', 'pointer');
+    }
+  }
+
+  function handleRectHover(e, rect) {
+    select(this)
+      .transition()
+      .attr('fill', keyGetter('hoverColor'))
+      .duration(150);
+
+    if (rect.tooltipText) {
+      createTooltip(rect);
+    }
+  }
+
+  function createTooltip(rect) {
+    const xPadding = 5;
+    const yPadding = 2;
+    const textY = rect.y - 8;
+    const textX = rect.x + xPadding;
+
+    const [bgId, textId, triangleId] = [
```

Comment:
destructuring a bit expensive, and you have even 5 lines instead of 3, or?

Reply:
I didn't want to write similar const 3 times)

Reply:
However, you did this before )
Const is more readable